### PR TITLE
CONSOLE-2524: Add Kata container RuntimeClass to workload detail pages

### PR DESCRIPTION
**Pull request by dtaylor113**
[CONSOLE-2524](https://issues.redhat.com/browse/CONSOLE-2524)

If you install the [Kata Operator](https://github.com/openshift/sandboxed-containers-operator/blob/release-4.7/README.md) on an OpenShift cluster you can then specify Deployment, DeploymentConfigs, ReplicaSets, etc. to create Pods using Kata containers.  For example, update a Deployment YAML to include `runtimeClassName`:

```
apiVersion: apps/v1
kind: Deployment
spec:
  template:
    spec:
      runtimeClassName: kata # ---> This can be changed
```
This will cause the Deployment to create Pods using kata containers which is shown in Pod YAML:
```
apiVersion: v1 
kind: Pod 
ginx-runc 
spec:     
  runtimeClassName: kata # ---> This cannot be changed here, it is generated
```

This PR addresses JIRA story [Add Kata runtime class property to workload details pages](https://issues.redhat.com/browse/CONSOLE-2524) and is part of a larger epic [Add Basic Visibility for Kata in Openshift UI](https://issues.redhat.com/browse/CONSOLE-2514)

The following workload Detail pages now have `RuntimeClass` field:

- Pods
- ReplicaSets
- ReplicationControllers
- StatefulSets
- Deployments
- DeploymentConfigs

### Pod Details page
image

### InfoTip
image

### Misc 
- Verified that `runtimeClassName` appears for Deployments and DeploymentConfigurations in Topology panel in Developer perspective.
- If `runtimeClassName` is not set for a particular resource, the field will be hidden on the details page.

### CODE REVIEW

File: frontend/public/components/replication-controller.jsx
Comment (rhamilto) [inline frontend/public/components/replication-controller.jsx:123]: Hooray for fixing invalid markup!

File: frontend/public/components/utils/details-page.tsx
Comment (spadgett) [inline frontend/public/components/utils/details-page.tsx:186]: I'd just call this `path` to be consistent with `DetailsItem` and to avoid confusion with CSS `className` properties.

```suggestion
export const RuntimeClass: React.FC<RuntimeClassProps> = ({ resource, path }) => {
```

File: frontend/public/components/utils/details-page.tsx
Comment (spadgett) [inline frontend/public/components/utils/details-page.tsx:192]:
```suggestion
      path={pathToClassName || 'spec.template.spec.runtimeClassName'}
```

File: frontend/public/components/utils/details-page.tsx
Comment (spadgett) [inline frontend/public/components/utils/details-page.tsx:190]:
```suggestion
      label={t('public~Runtime class')}
```

File: frontend/public/components/utils/details-page.tsx
Comment (spadgett) [reply frontend/public/components/utils/details-page.tsx:186]: I'd also have a preference for calling the prop `obj` instead of `resource` to be consistent with `DetailsItem`. `obj` also is more clearly a specific instance whereas `resource` sometimes refers to a type of resource in k8s.

File: frontend/public/components/utils/details-page.tsx
Comment (spadgett) [inline frontend/public/components/utils/details-page.tsx:216]: Prefer `K8sResourceCommon` in most places, which has stricter type checking.

```suggestion
  resource: K8sResourceCommon;
```

**Review by spadgett — COMMENTED** [review]
Thanks, @dtaylor113. Looks good.

**Comment by c3d** [comment]
Thanks @dtaylor113.

**Comment by c3d** [comment]
@dtaylor113 Would it make sense for the infotip to mention `kata` as an example of valid value for the `runtimeClass`?

**Comment by zanetworker** [comment]
@c3d is kata the final immutable name `runtimeClass` name? what if we decide to change it to `kata-qemu`, `kata-visor`,... That would require a PR or is there another way?

**Comment by c3d** [comment]
> @c3d is kata the final immutable name `runtimeClass` name? what if we decide to change it to `kata-qemu`, `kata-visor`,... That would require a PR or is there another way?

For now, we have decided to keep the runtime class name as `kata`. There can be more than one runtime class name. For example, we could have a "kata-debug" that has more debug information.

**Comment by zanetworker** [comment]
@c3d Then IMO I suggest we don't hard-code the name in the info unless there is a way to add more names

File: frontend/public/components/utils/details-page.tsx
Comment (dtaylor113) [reply frontend/public/components/utils/details-page.tsx]: fixed

File: frontend/public/components/utils/details-page.tsx
Comment (dtaylor113) [reply frontend/public/components/utils/details-page.tsx]: fixed

File: frontend/public/components/utils/details-page.tsx
Comment (dtaylor113) [reply frontend/public/components/utils/details-page.tsx]: fixed

File: frontend/public/components/utils/details-page.tsx
Comment (dtaylor113) [reply frontend/public/components/utils/details-page.tsx]: Fixed,  really wish I could test these last fixes, trying to locate a cluster which has Kata Runtime/Example installed on it.

**Review by spadgett — APPROVED** [review]
/lgtm
/hold for approvals
/assign @yapei @ahardin-rh @sferich888

**Comment by yapei** [comment]
Confirmed that `Runtime class` field is shown on Pod/Deployment/DeploymentConfig/ReplicaSet/ReplicationController/StatefulSet detail page

The only issue is that on Deployment/ReplicaSet/ReplicationController details page, I didn't see a popover help for `Runtime class` field as other detail pages, this may not in this PR's scope
Screen Shot 2021-03-19 at 4 41 30 PM

**Comment by dtaylor113** [comment]
@yapei , hmm...I'm seeing popover help on Deployment/ReplicaSet/ReplicationController details page:
image
image
image

Can you pls test again?
- thanks

**Comment by dtaylor113** [comment]
@yapei wrote: "[to be] clear in comment. Runtime class  popover is clickable on all resources detail page.  The minor issue is  the dotted line ______  is not shown for Runtime class  when comparing with other fields"

Investigating...

**Comment by yapei** [comment]
@dtaylor113  I tested again on Chrome 89 and Firefox 86, I do see the dotted line this time on Deployment/ReplicaSet/ReplicationController details page

**Review by andrew-ronaldson — APPROVED** [review]
looks great! Thanks @dtaylor113

**Comment by dtaylor113** [comment]
Hi @sferich888, when you get a moment, waiting on "px-approved" label.  -thanks

**Comment by cpmeadors** [comment]
I am looking at an OCP cluster build with 4.8.0-0.nightly-2021-03-30-123133 and I don't see the "Runtime class" in any of the expected locations for any of my pods/deployments using "runtimeClassName: kata".  What am I missing?

**Comment by dtaylor113** [comment]
> I am looking at an OCP cluster build with 4.8.0-0.nightly-2021-03-30-123133 and I don't see the "Runtime class" in any of the expected locations for any of my pods/deployments using "runtimeClassName: kata". What am I missing?

Hi @cpmeadors, a 4.8 nightly cluster won't show "Runtime class" until this PR is merged.

**Pull request merged**
